app/controllers/TaskController.js

```diff
@@ -0,0 +1,771 @@
+   req.checkBody('title', 'required').notEmpty();
+   req.checkBody('title', 'validity').isString();
+   req.sanitizeBody('title').escape();
+   req.sanitizeBody('title').trim();
+
+   /*Validate and sanitizing description Input*/
+   if(req.body.description)
+   {
+     req.checkBody('description', 'validity').isString();
+     req.sanitizeBody('description').escape();
+     req.sanitizeBody('description').trim();
+   }
+   else
+   {
+     req.body.description = null;
+   }
+
+   /*Validate and sanitizing deadline Input*/
+   req.checkBody('deadline', 'required').notEmpty();
+   req.checkBody('deadline', 'validity').isDate();
+   req.sanitizeBody('deadline').escape();
+   req.sanitizeBody('deadline').trim();
+
+   if(req.body.assigned_to)
+     req.checkBody('assigned_to', 'validity').isArray();
+
+   /*Validate and sanitizing end date Input*/
+   if(req.body.evaluation)
+   {
+      req.sanitizeBody('evaluation').escape();
+      req.sanitizeBody('evaluation').trim();
+   }
+   else
+      req.body.evaluation = null;
+
+   /*Validate and sanitizing end date Input*/
+   req.checkBody('priority', 'required').notEmpty();
+   req.sanitizeBody('priority').escape();
+   req.sanitizeBody('priority').trim();
+
+   var p = req.body.priority;
+   var errors = req.validationErrors();
+   var assigned_to = null;
+
+   if(p!=1 && p!= 3 && p!= 5 && p!= 8)
+   {
+     if(!errors)
+      errors = [];
+
+     errors.push({
+        param: 'priority',
+        value: req.body.priority,
+        msg: 'validity'
+     });
+   }
+
+   var rest = function()
+   {
+      /*sending validation errors*/
+      errors = format(errors);
+      if (errors)
+      {
+         /* input validation failed */
+         res.status(400).json({
+            status: 'failed',
+            errors: errors
+         });
+
+         req.err = 'TaskController.js, Line: 238\nSome validation errors occurred.\n' + JSON.stringify(errors);
+         next();
+         return;
+      }
+
+      var attributes =
+      {
+         title: req.body.title,
+         description: req.body.description,
+         deadline: req.body.deadline,
+         priority: req.body.priority,
+         status: 'New',
+         evaluation: req.body.evaluation,
+         supervisor: req.user.id
+      };
+
+      Task.create(attributes).then(function(task)
+       {
+         task.setAssignedUsers(req.body.assigned_to).then(function(){
+            res.status(200).json({
+               status: 'succeeded',
+               message: 'task successfully created'
+            });
+
+            next();
+            return;
+         }).catch(function(err) {
+            /* failed to update the task assigned_to in the database */
+            res.status(500).json({
+               status:'failed',
+               message: 'Internal server error'
+            });
+
+            req.err = 'TaskController.js, Line: 470\nfailed to update the task assigned_to in the database.\n' + String(err);
+            next();
```

Comment:
here destroy the task before as you have added the task and didn't add the assignees